src/Avalonia.Base/Threading/Dispatcher.cs

```diff
@@ -88,4 +88,32 @@ internal AvaloniaSynchronizationContext GetContextWithPriority(DispatcherPriorit
         var index = priority - DispatcherPriority.MinValue;
         return _priorityContexts[index] ??= new(priority);
     }
+
+    /// <summary>
+    /// Creates an awaitable object that asynchronously yields control back to the current dispatcher
+    /// and provides an opportunity for the dispatcher to process other events.
+    /// </summary>
+    /// <returns>
+    /// An awaitable object that asynchronously yields control back to the current dispatcher
+    /// and provides an opportunity for the dispatcher to process other events.
+    /// </returns>
+    /// <remarks>
+    /// This method is equivalent to calling the <see cref="Yield(DispatcherPriority)"/> method
+    /// and passing in <see cref="DispatcherPriority.Background"/>.
+    /// </remarks>
+    public DispatcherYieldAwaitable Yield() =>
```

Comment:
Can we have WPF's naming? `DispatcherPriorityAwaitable`/`DispatcherPriorityAwaiter`

Reply:
Yes, if it's okay to rewrite `AwaitWithPriority` methods like in the following snippet and repurpose the existing `DispatcherPriorityAwaitable`.

```csharp
public async Task AwaitWithPriority(Task task, DispatcherPriority priority)
{
    await Yield(priority);
    await task;
}

public async Task<T> AwaitWithPriority<T>(Task<T> task, DispatcherPriority priority)
{
    await Yield(priority);
    return await task;
}
```

`Yield` should go first because the current behavior is to continue any task on the UI thread even if it's failed. Setting the current synchronization context is a solution too, but then the `task` must be awaited anyway to capture the context.